File: swift_mort/MORT/Models/Proof.swift
import Foundation

struct ProofUpload: Codable, Hashable, Identifiable, Sendable {
    let id: UUID
    let applicationID: UUID
    let uploadedBy: UUID
    let storagePath: String
    let note: String?
    let status: String
    let reviewedBy: UUID?
    let reviewNote: String?
    let reviewedAt: String?
    let createdAt: String?
    let updatedAt: String?

    enum CodingKeys: String, CodingKey {
        case id, note, status
        case applicationID = "application_id"
        case uploadedBy = "uploaded_by"
        case storagePath = "storage_path"
        case reviewedBy = "reviewed_by"
        case reviewNote = "review_note"
        case reviewedAt = "reviewed_at"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }

    var statusTitle: String {
        switch status {
        case "submitted": "Awaiting review"
        case "approved": "Approved"
        case "resubmission_requested": "New proof requested"
        case "rejected": "Needs attention"
        default: status.replacingOccurrences(of: "_", with: " ").capitalized
        }
    }
}

File: flutter_mort/lib/data/models/proof.dart
class ProofUpload {
  const ProofUpload({
    required this.id,
    required this.applicationId,
    required this.uploadedBy,
    required this.storagePath,
    required this.status,
    this.note,
    this.reviewedBy,
    this.reviewNote,
    this.reviewedAt,
    this.createdAt,
    this.updatedAt,
  });

  final String id;
  final String applicationId;
  final String uploadedBy;
  final String storagePath;
  final String status;
  final String? note;
  final String? reviewedBy;
  final String? reviewNote;
  final DateTime? reviewedAt;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  String get statusLabel => switch (status) {
    'submitted' => 'Awaiting review',
    'approved' => 'Approved',
    'resubmission_requested' => 'New proof requested',
    'rejected' => 'Needs attention',
    _ => status.replaceAll('_', ' '),
  };

  factory ProofUpload.fromMap(Map<String, dynamic> json) {
    return ProofUpload(
      id: json['id'].toString(),
      applicationId: json['application_id'].toString(),
      uploadedBy: json['uploaded_by'].toString(),
      storagePath: (json['storage_path'] as String?) ?? '',
      status: (json['status'] as String?) ?? 'submitted',
      note: json['note'] as String?,
      reviewedBy: json['reviewed_by'] as String?,
      reviewNote: json['review_note'] as String?,
      reviewedAt: DateTime.tryParse((json['reviewed_at'] ?? '').toString()),
      createdAt: DateTime.tryParse((json['created_at'] ?? '').toString()),
      updatedAt: DateTime.tryParse((json['updated_at'] ?? '').toString()),
    );
  }
}
